### PR TITLE
Setup cri-tools for docker

### PR DESCRIPTION
**Pull request by mrunalp**
This should be setup when docker is installed but cri-o isn't.

@sdodson PTAL

Signed-off-by: Mrunal Patel <mrunalp@gmail.com>

### CODE REVIEW

File: roles/container_runtime/tasks/package_docker.yml [hunk 1]
```diff
@@ -31,6 +31,20 @@
...
+    state: present
+
+- name: Configure cri-tools for dockershim
+  copy:
+    contents: |
+      runtime-endpoint: unix:///var/run/dockershim.sock
+      image-endpoint: unix:///var/run/dockershim.sock
+    dest: /etc/crictl.yaml
+  when:
+  - not (openshift_use_crio | bool)
```
Comment (mrunalp) [inline roles/container_runtime/tasks/package_docker.yml]: @sdodson I am not sure whether this is the right way. We want this to be installed on masters even when nodes are expected to have crio installed.

Comment (mrunalp) [reply roles/container_runtime/tasks/package_docker.yml]: Do you have suggestions?

File: roles/container_runtime/tasks/package_docker.yml [hunk 2]
```diff
@@ -31,6 +31,20 @@
   register: result
   until: result is succeeded
 
+ - name: Install cri-tools
+   package:
+    name: "cri-tools"
+    state: present
```
Comment (sdodson) [inline roles/container_runtime/tasks/package_docker.yml]: Any reason not to say latest? This ships in the per release repos, right?

Comment (mrunalp) [reply roles/container_runtime/tasks/package_docker.yml]: Sure, I'll fix that

File: roles/container_runtime/tasks/package_docker.yml (hunk 1 above)
Comment (sdodson) [reply roles/container_runtime/tasks/package_docker.yml]: That will work as long as that's all that you'd want in the file. If the user may add other values that are unrelated we should use http://docs.ansible.com/ansible/lineinfile_module.html

File: roles/container_runtime/tasks/package_docker.yml (hunk 2 above)
Comment (mrunalp) [reply roles/container_runtime/tasks/package_docker.yml]: Fixed.

**Comment by sdodson** [comment]
Just one general thing to be aware of, cri-tools is not installed on atomic host. So it's difficult to make everything depend on crictl if that's a goal you're working towards.

File: roles/container_runtime/tasks/package_docker.yml (hunk 1 above)
Comment (mrunalp) [reply roles/container_runtime/tasks/package_docker.yml]: okay! yeah, that's what we would want in that file as long as crio isn't installed as well. So this is for use cases where we run docker on the masters and crio on the nodes for example.

**Comment by mrunalp** [comment]
@ashcrow Can we get cri-tools installed on atomic?

File: roles/container_runtime/tasks/package_docker.yml (hunk 1 above)
Comment (sdodson) [reply roles/container_runtime/tasks/package_docker.yml]: sounds good

**Comment by ashcrow** [comment]
@mrunalp It's possible yes. However, ``cri-o`` runs as a system container on Atomic Host. We could add ``cri-tools`` to a future Atomic compose if it's required. Otherwise I'd recommend adding it into the system container.

File: roles/container_runtime/tasks/package_docker.yml [hunk 3]
```diff
@@ -31,6 +31,20 @@
   register: result
   until: result is succeeded
 
+ - name: Install cri-tools
+   package:
+    name: "cri-tools"
+    state: latest
```
[the remark this answers is not in the document]
Comment (mrunalp) [reply roles/container_runtime/tasks/package_docker.yml]: fixed.

**Comment by sdodson** [comment]
> @mrunalp It's possible yes. However, cri-o runs as a system container on Atomic Host. We could add cri-tools to a future Atomic compose if it's required. Otherwise I'd recommend adding it into the system container.

Since crio was found not to work well as a system container we don't currently support crio on atomic host in anyway. Maybe that's been fixed, I don't know, but we'd decided just to wait on RHCOS.

**Comment by ashcrow** [comment]
> [...] I don't know, but we'd decided just to wait on RHCOS.

We can totally add ``cri-tools`` to RHCOS. Follow up with us via email and let us know where to grab the package from and we can add it to the composes.

**Comment by sdodson** [comment]
@mrunalp @runcom Is this a blocker for 3.10?

**Comment by mrunalp** [comment]
@sdodson Yes, we want this for 3.10.

**Comment by sdodson** [comment]
/close
no further updates to 3.10

**Pull request closed without merging**